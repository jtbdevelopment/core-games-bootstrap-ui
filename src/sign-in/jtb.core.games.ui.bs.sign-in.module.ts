import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {NgbModule} from '@ng-bootstrap/ng-bootstrap';
import {SignInComponent} from './sign-in.component';
import {JTBCoreGamesUI} from 'jtb-core-games-ui';

@NgModule({
    imports: [
        BrowserModule,
        NgbModule,
        JTBCoreGamesUI
    ],
    providers: [
        {provide: 'Window', useValue: window}
    ],
    exports: [
        SignInComponent
    ],
    declarations: [
        SignInComponent
    ]
})
export class JTBCoreGamesUIBSSignInModule {
}

