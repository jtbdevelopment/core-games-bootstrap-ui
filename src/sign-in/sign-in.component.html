<div class="sign-in">
    <div id="sign-in-message" class="container-fluid">
        <div class="row text-center">
            <div class="col">
                <p>{{message}}</p>
            </div>
        </div>
    </div>
    <div id="sign-in-facebook" class="container-fluid" *ngIf="showFacebook">
        <div class="row text-center">
            <div class="col">
                <a href="javascript:void(0)" id="fb-login" (click)="fbLogin()"><img
                        src="../../../images/sign-in-with-facebook.png"/></a>
            </div>
        </div>
    </div>
    <div id="sign-in-manual" class="container-fluid" *ngIf="showManual">
        <div class="row text-center">
            <div class="col">
                <p>Don't want to login with Facebook?</p>
                <!--suppress HtmlUnknownTarget -->
                <form id="signin" action="/signin/authenticate" method="post" class="form-group">
                    <div class="form-group">
                        <label for="username">Username</label>
                        <input id="username" name="username" type="text" size="25"/>
                    </div>
                    <div class="form-group">
                        <label for="password">Password</label>
                        <input id="password" name="password" type="password" size="25"/>
                    </div>
                    <div class="form-group">
                        <div class="checkbox">
                            <label>
                                <input type="checkbox" id="remember-me" name="remember-me"> Remember me
                            </label>
                        </div>
                    </div>
                    <div class="form-group">
                        <button type="submit" class="btn btn-default">Sign in</button>
                    </div>
                </form>
            </div>
        </div>
    </div>
</div>
