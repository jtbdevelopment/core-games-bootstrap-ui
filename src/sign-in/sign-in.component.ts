import {Component, Inject} from '@angular/core';
import {FacebookInitializerService, FacebookLoginService} from 'jtb-core-games-ui';

@Component({
    selector: 'sign-in',
    template: require('./sign-in.component.html'),
})
export class SignInComponent {
    public message: string = '';
    public showManual: boolean = false;
    public showFacebook: boolean = false;

    constructor(private facebookLogin: FacebookLoginService,
                private facebookInit: FacebookInitializerService,
                @Inject('Window') private window: Window) {
        this.showManual = true;
        this.showFacebook = false;
        this.facebookInit.fbReady.then(() => {
            this.showFacebook = true;
        });
        this.facebookLogin.canAutoLogin.subscribe((can: boolean) => {
            if (can) {
                this.autoLogin();
            }
        });
    }

    public fbLogin(): void {
        this.facebookLogin.initiateLogin();
    }

    private autoLogin(): void {
        this.showFacebook = false;
        this.showManual = false;
        this.message = 'Logging in via Facebook';
        this.window.location.href = '/auth/facebook';
    }
}
