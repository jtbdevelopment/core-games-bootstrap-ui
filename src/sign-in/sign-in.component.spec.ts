import {ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
import {SignInComponent} from './sign-in.component';
import {BehaviorSubject} from 'rxjs/BehaviorSubject';
import {FacebookInitializerService, FacebookLoginService} from 'jtb-core-games-ui';

class MockFacebookLogin {
    public canAutoLogin: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);

    initiateLogin = jasmine.createSpy('initLogin');
}

class MockFacebookInit {
    public resolve: (result?: boolean) => void;
    public reject: (reason?: any) => void;
    public fbReady: Promise<any> = new Promise<any>((resolve, reject) => {
        this.resolve = resolve;
        this.reject = reject;
    });
}

class MockLocation {
    public href: string = '';
}

class MockWindow {
    public location: MockLocation = new MockLocation();
}

describe('Component:  sign in component', () => {
    let fbLogin: MockFacebookLogin;
    let fbInit: MockFacebookInit;
    let mockWindow: MockWindow;
    let fixture: ComponentFixture<SignInComponent>;

    beforeEach(() => {
        TestBed.configureTestingModule({
            providers: [
                {provide: FacebookLoginService, useClass: MockFacebookLogin},
                {provide: FacebookInitializerService, useClass: MockFacebookInit},
                {provide: 'Window', useClass: MockWindow}
            ],
            declarations: [
                SignInComponent,
            ]
        });
        TestBed.compileComponents();
        fixture = TestBed.createComponent(SignInComponent);
        fbLogin = TestBed.get(FacebookLoginService);
        fbInit = TestBed.get(FacebookInitializerService);
        mockWindow = TestBed.get('Window');
    });

    describe('non manual fb logins', () => {
        afterEach(() => {
            expect(fbLogin.initiateLogin).not.toHaveBeenCalled();
        });

        it('initially displays manual only', () => {
            fixture.detectChanges();
            const screen = fixture.nativeElement;
            expect(screen.querySelector('#sign-in-message')).toBeDefined();
            expect(screen.querySelector('#sign-in-message')).not.toBeNull();
            expect(screen.querySelector('#sign-in-facebook')).toBeNull();
            expect(screen.querySelector('#sign-in-manual')).toBeDefined();
            expect(screen.querySelector('#sign-in-manual')).not.toBeNull();
            expect(screen.querySelector('#sign-in-message').textContent.trim()).toEqual('');
        });

        it('displays facebook after init', fakeAsync(() => {
            fbInit.resolve(true);
            tick();
            fixture.detectChanges();
            const screen = fixture.nativeElement;
            expect(screen.querySelector('#sign-in-message')).toBeDefined();
            expect(screen.querySelector('#sign-in-facebook')).toBeDefined();
            expect(screen.querySelector('#sign-in-manual')).toBeDefined();
            expect(screen.querySelector('#sign-in-message').textContent.trim()).toEqual('');
        }));

        it('initiates auto login if possible', () => {
            fbLogin.canAutoLogin.next(true);
            expect(mockWindow.location.href).toEqual('/auth/facebook');
            const screen = fixture.nativeElement;
            expect(screen.querySelector('#sign-in-message')).toBeDefined();
            expect(screen.querySelector('#sign-in-facebook')).toBeNull();
            expect(screen.querySelector('#sign-in-manual')).toBeNull();
            expect(screen.querySelector('#sign-in-message').textContent.trim()).toEqual('');
        });

        it('Does not initiate auto login if not possible', () => {
            fbLogin.canAutoLogin.next(false);
            expect(mockWindow.location.href).toEqual('');
            const screen = fixture.nativeElement;
            expect(screen.querySelector('#sign-in-message')).toBeDefined();
            expect(screen.querySelector('#sign-in-facebook')).toBeDefined();
            expect(screen.querySelector('#sign-in-manual')).toBeDefined();
            expect(screen.querySelector('#sign-in-message').textContent.trim()).toEqual('');
        });

        it('displays message', () => {
            let m = 'A Message!';
            fixture.componentInstance.message = m;
            fixture.detectChanges();
            const screen = fixture.nativeElement;
            expect(screen.querySelector('#sign-in-message').textContent.trim()).toEqual(m);
        });
    });

    it('clicking fb initiates manual fb login', fakeAsync(() => {
        fbInit.resolve(true);
        tick();
        fixture.detectChanges();
        const screen = fixture.nativeElement;
        expect(screen.querySelector('#sign-in-facebook')).toBeDefined();
        let fbButton = screen.querySelector('#fb-login');
        expect(fbButton).toBeDefined();
        expect(fbButton).not.toBeNull();
        fbButton.click();
        expect(fbLogin.initiateLogin).toHaveBeenCalled();
    }));
});
