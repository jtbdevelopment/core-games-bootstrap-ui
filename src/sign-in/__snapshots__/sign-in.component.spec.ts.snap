// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  sign in component clicking fb initiates manual fb login 1`] = `
<sign-in
  facebookInit={[Function MockFacebookInit]}
  facebookLogin={[Function MockFacebookLogin]}
  message=""
  showFacebook={[Function Boolean]}
  showManual={[Function Boolean]}
>
  <div
    class="sign-in"
  >
    
    
    <div
      class="container-fluid"
      id="sign-in-message"
    >
      
      
      <div
        class="row text-center"
      >
        
        
        <div
          class="col"
        >
          
          
          <p>
            
          </p>
          
        
        </div>
        
      
      </div>
      
    
    </div>
    
    
    
    <div
      class="container-fluid"
      id="sign-in-facebook"
    >
      
      
      <div
        class="row text-center"
      >
        
        
        <div
          class="col"
        >
          
          
          <a
            href="javascript:void(0)"
            id="fb-login"
          >
            <img
              src="/assets/sign-in-with-facebook.png"
            />
          </a>
          
        
        </div>
        
      
      </div>
      
    
    </div>
    
    
    
    <div
      class="container-fluid"
      id="sign-in-manual"
    >
      
      
      <div
        class="row text-center"
      >
        
        
        <div
          class="col"
        >
          
          
          <p>
            Don't want to login with Facebook?
          </p>
          
          
          
          
          <form
            action="/signin/authenticate"
            class="form-group"
            id="signin"
            method="post"
          >
            
            
            <div
              class="form-group"
            >
              
              
              <label
                for="username"
              >
                Username
              </label>
              
              
              <input
                id="username"
                name="username"
                size="25"
                type="text"
              />
              
            
            </div>
            
            
            <div
              class="form-group"
            >
              
              
              <label
                for="password"
              >
                Password
              </label>
              
              
              <input
                id="password"
                name="password"
                size="25"
                type="password"
              />
              
            
            </div>
            
            
            <div
              class="form-group"
            >
              
              
              <div
                class="checkbox"
              >
                
                
                <label>
                  
                  
                  <input
                    id="remember-me"
                    name="remember-me"
                    type="checkbox"
                  />
                   Remember me
                
                </label>
                
              
              </div>
              
            
            </div>
            
            
            <div
              class="form-group"
            >
              
              
              <button
                class="btn btn-default"
                type="submit"
              >
                Sign in
              </button>
              
            
            </div>
            
          
          </form>
          
        
        </div>
        
      
      </div>
      
    
    </div>
    
  
  </div>
  
</sign-in>
`;

exports[`Component:  sign in component non manual fb logins Does not initiate auto login if not possible 1`] = `
<sign-in
  facebookInit={[Function MockFacebookInit]}
  facebookLogin={[Function MockFacebookLogin]}
  message=""
  showFacebook="false"
  showManual={[Function Boolean]}
>
  <div
    class="sign-in"
  >
    
    
    <div
      class="container-fluid"
      id="sign-in-message"
    >
      
      
      <div
        class="row text-center"
      >
        
        
        <div
          class="col"
        >
          
          
          <p>
            
          </p>
          
        
        </div>
        
      
      </div>
      
    
    </div>
    
    
    
    
    
    
    
  
  </div>
  
</sign-in>
`;

exports[`Component:  sign in component non manual fb logins displays facebook after init 1`] = `
<sign-in
  facebookInit={[Function MockFacebookInit]}
  facebookLogin={[Function MockFacebookLogin]}
  message=""
  showFacebook={[Function Boolean]}
  showManual={[Function Boolean]}
>
  <div
    class="sign-in"
  >
    
    
    <div
      class="container-fluid"
      id="sign-in-message"
    >
      
      
      <div
        class="row text-center"
      >
        
        
        <div
          class="col"
        >
          
          
          <p>
            
          </p>
          
        
        </div>
        
      
      </div>
      
    
    </div>
    
    
    
    <div
      class="container-fluid"
      id="sign-in-facebook"
    >
      
      
      <div
        class="row text-center"
      >
        
        
        <div
          class="col"
        >
          
          
          <a
            href="javascript:void(0)"
            id="fb-login"
          >
            <img
              src="/assets/sign-in-with-facebook.png"
            />
          </a>
          
        
        </div>
        
      
      </div>
      
    
    </div>
    
    
    
    <div
      class="container-fluid"
      id="sign-in-manual"
    >
      
      
      <div
        class="row text-center"
      >
        
        
        <div
          class="col"
        >
          
          
          <p>
            Don't want to login with Facebook?
          </p>
          
          
          
          
          <form
            action="/signin/authenticate"
            class="form-group"
            id="signin"
            method="post"
          >
            
            
            <div
              class="form-group"
            >
              
              
              <label
                for="username"
              >
                Username
              </label>
              
              
              <input
                id="username"
                name="username"
                size="25"
                type="text"
              />
              
            
            </div>
            
            
            <div
              class="form-group"
            >
              
              
              <label
                for="password"
              >
                Password
              </label>
              
              
              <input
                id="password"
                name="password"
                size="25"
                type="password"
              />
              
            
            </div>
            
            
            <div
              class="form-group"
            >
              
              
              <div
                class="checkbox"
              >
                
                
                <label>
                  
                  
                  <input
                    id="remember-me"
                    name="remember-me"
                    type="checkbox"
                  />
                   Remember me
                
                </label>
                
              
              </div>
              
            
            </div>
            
            
            <div
              class="form-group"
            >
              
              
              <button
                class="btn btn-default"
                type="submit"
              >
                Sign in
              </button>
              
            
            </div>
            
          
          </form>
          
        
        </div>
        
      
      </div>
      
    
    </div>
    
  
  </div>
  
</sign-in>
`;

exports[`Component:  sign in component non manual fb logins displays message 1`] = `
<sign-in
  facebookInit={[Function MockFacebookInit]}
  facebookLogin={[Function MockFacebookLogin]}
  message={[Function String]}
  showFacebook="false"
  showManual={[Function Boolean]}
>
  <div
    class="sign-in"
  >
    
    
    <div
      class="container-fluid"
      id="sign-in-message"
    >
      
      
      <div
        class="row text-center"
      >
        
        
        <div
          class="col"
        >
          
          
          <p>
            A Message!
          </p>
          
        
        </div>
        
      
      </div>
      
    
    </div>
    
    
    
    
    
    
    <div
      class="container-fluid"
      id="sign-in-manual"
    >
      
      
      <div
        class="row text-center"
      >
        
        
        <div
          class="col"
        >
          
          
          <p>
            Don't want to login with Facebook?
          </p>
          
          
          
          
          <form
            action="/signin/authenticate"
            class="form-group"
            id="signin"
            method="post"
          >
            
            
            <div
              class="form-group"
            >
              
              
              <label
                for="username"
              >
                Username
              </label>
              
              
              <input
                id="username"
                name="username"
                size="25"
                type="text"
              />
              
            
            </div>
            
            
            <div
              class="form-group"
            >
              
              
              <label
                for="password"
              >
                Password
              </label>
              
              
              <input
                id="password"
                name="password"
                size="25"
                type="password"
              />
              
            
            </div>
            
            
            <div
              class="form-group"
            >
              
              
              <div
                class="checkbox"
              >
                
                
                <label>
                  
                  
                  <input
                    id="remember-me"
                    name="remember-me"
                    type="checkbox"
                  />
                   Remember me
                
                </label>
                
              
              </div>
              
            
            </div>
            
            
            <div
              class="form-group"
            >
              
              
              <button
                class="btn btn-default"
                type="submit"
              >
                Sign in
              </button>
              
            
            </div>
            
          
          </form>
          
        
        </div>
        
      
      </div>
      
    
    </div>
    
  
  </div>
  
</sign-in>
`;

exports[`Component:  sign in component non manual fb logins initially displays manual only 1`] = `
<sign-in
  facebookInit={[Function MockFacebookInit]}
  facebookLogin={[Function MockFacebookLogin]}
  message=""
  showFacebook="false"
  showManual={[Function Boolean]}
>
  <div
    class="sign-in"
  >
    
    
    <div
      class="container-fluid"
      id="sign-in-message"
    >
      
      
      <div
        class="row text-center"
      >
        
        
        <div
          class="col"
        >
          
          
          <p>
            
          </p>
          
        
        </div>
        
      
      </div>
      
    
    </div>
    
    
    
    
    
    
    <div
      class="container-fluid"
      id="sign-in-manual"
    >
      
      
      <div
        class="row text-center"
      >
        
        
        <div
          class="col"
        >
          
          
          <p>
            Don't want to login with Facebook?
          </p>
          
          
          
          
          <form
            action="/signin/authenticate"
            class="form-group"
            id="signin"
            method="post"
          >
            
            
            <div
              class="form-group"
            >
              
              
              <label
                for="username"
              >
                Username
              </label>
              
              
              <input
                id="username"
                name="username"
                size="25"
                type="text"
              />
              
            
            </div>
            
            
            <div
              class="form-group"
            >
              
              
              <label
                for="password"
              >
                Password
              </label>
              
              
              <input
                id="password"
                name="password"
                size="25"
                type="password"
              />
              
            
            </div>
            
            
            <div
              class="form-group"
            >
              
              
              <div
                class="checkbox"
              >
                
                
                <label>
                  
                  
                  <input
                    id="remember-me"
                    name="remember-me"
                    type="checkbox"
                  />
                   Remember me
                
                </label>
                
              
              </div>
              
            
            </div>
            
            
            <div
              class="form-group"
            >
              
              
              <button
                class="btn btn-default"
                type="submit"
              >
                Sign in
              </button>
              
            
            </div>
            
          
          </form>
          
        
        </div>
        
      
      </div>
      
    
    </div>
    
  
  </div>
  
</sign-in>
`;
